src/amici_interface_matlab.cpp

```diff
@@ -94,19 +100,19 @@ enum mexRhsArguments {
  * userDataFromMatlabCall parses the input from the matlab call and writes it to
  * an UserData class object
  *
- * @param[in] nrhs number of input arguments of the matlab call @type int
- * @param[in] prhs pointer to the array of input arguments @type mxArray
- * @param[in] model pointer to the model object, this is necessary to perform
- * dimension checks @type Model
- * @return udata pointer to user data object @type UserData
+ * @param nrhs number of input arguments of the matlab call
+ * @param prhs pointer to the array of input arguments
+ * @return udata pointer to user data object
  */
-UserData *userDataFromMatlabCall(const mxArray *prhs[], int nrhs,
-                                 Model *model) {
+template <class mxArray>
+UserData *userDataFromMatlabCall(const mxArray *prhs[], int nrhs) {
```

Comment:
Similar as above: I think passing the model instance is much clearer than using the model wrapper functions.

Reply:
see above